### PR TITLE
REMO-12: Implement draggable item

### PR DESCRIPTION
**Pull request by chalalala**
## Changelog
- Added package `react-beautiful-dnd` which supports implementing drag-and-drop items
- Implemented draggable item
- Updated new section ID

## Recording
[Screencast from 22-10-2023 14:49:51.webm](https://github.com/chalalala/remo/assets/49138712/b785e188-38a0-45a6-b67a-b9432ac318a4)

### CODE REVIEW

File: components/DraggableItem/DraggableItem.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  const Wrapper = url
+    ? ({ children, ...props }: PropsWithChildren) => (
+        <a
+          href={url}
+          {...props}
+        >
+          {children}
+        </a>
+      )
+    : 'span';
```
Comment (thuutri2710) [inline components/DraggableItem/DraggableItem.tsx:22]: it's better if u move this component outside `DraggableItem`

File: utils/sectionList.ts [hunk 2]
```diff
@@ -38,10 +39,44 @@ export const addSection = (sections: Section[], name: string) => {
...
+  )?.[0];
+
+  modifiedSections[destinationSectionIdx].items.splice(
+    dropResult.destination.index,
+    0,
+    draggedItem,
+  );
+
+  return modifiedSections;
+};
```
Comment (thuutri2710) [inline utils/sectionList.ts:82]: we should write test for this function

**Review by thuutri2710 — CHANGES_REQUESTED** [review]

File: utils/sectionList.ts (hunk 2 above)
Comment (chalalala) [reply utils/sectionList.ts:82]: Tests are already included here
https://github.com/chalalala/remo/blob/907322cedee976c917fad4c5bb902d19cb56c27b/utils/__tests__/sectionList.test.ts#L67-L218

File: components/DraggableItem/DraggableItem.tsx (hunk 1 above)
Comment (chalalala) [reply components/DraggableItem/DraggableItem.tsx]: Updated at [b494200](https://github.com/chalalala/remo/pull/9/commits/b49420008e1bf2b55bed619ffc8bf65ec783d3eb)

**Review by thuutri2710 — APPROVED** [review]
LGTM

**Pull request merged**